ffi/derive/src/derive.rs

```diff
@@ -0,0 +1,225 @@
+enum Derive {
+    Set,
+    Get,
+    GetMut,
+}
+
+/// Generate FFI function equivalents of derived methods
+pub fn gen_fns_from_derives(item: &ItemStruct) -> Vec<syn::ItemFn> {
+    let struct_derives = parse_derives(&item.attrs).unwrap_or_default();
+
+    let mut ffi_derives = vec![];
+    match &item.fields {
+        syn::Fields::Named(syn::FieldsNamed { named, .. }) => named.iter().for_each(|field| {
+            if let Some(mut field_derives) = parse_derives(&field.attrs) {
+                field_derives.extend(struct_derives.clone());
+
+                for derive in field_derives {
+                    ffi_derives.push(gen_ffi_derive(&item.ident, field, derive));
+                }
+            }
+        }),
+        syn::Fields::Unnamed(_) | syn::Fields::Unit => {
+            abort!(item, "Only named structs supported")
+        }
+    }
+
+    ffi_derives
+}
+
+/// Parses getset attributes to find out which methods it derives
+fn parse_derives(attrs: &[syn::Attribute]) -> Option<HashSet<Derive>> {
+    let mut getset_attrs = HashSet::new();
+
+    for attr in attrs {
+        if let Ok(syn::Meta::List(meta_list)) = attr.parse_meta() {
+            if !meta_list.path.is_ident("getset") {
+                continue;
+            }
+
+            for nested in meta_list.nested {
+                if let syn::NestedMeta::Meta(item) = nested {
+                    match item {
+                        syn::Meta::NameValue(item) => {
+                            if item.lit != parse_quote! { "pub" } {
+                                continue;
+                            }
+
+                            if item.path.is_ident("set") {
+                                getset_attrs.insert(Derive::Set);
+                            } else if item.path.is_ident("get") {
+                                getset_attrs.insert(Derive::Get);
+                            } else if item.path.is_ident("get_mut") {
+                                getset_attrs.insert(Derive::GetMut);
+                            }
+                        }
+                        syn::Meta::Path(path) => {
+                            if path.is_ident("skip") {
+                                return None;
+                            }
+                        }
+                        _ => abort!(item, "Unsupported getset attribute"),
+                    }
+                }
+            }
+        }
+    }
+
+    Some(getset_attrs)
+}
+
+fn gen_derive_method_name(field_name: &Ident, derive: Derive) -> syn::Ident {
+    Ident::new(
+        &match derive {
+            Derive::Set => format!("set_{}", field_name),
+            Derive::Get => format!("{}", field_name),
+            Derive::GetMut => format!("{}_mut", field_name),
+        },
+        proc_macro2::Span::call_site(),
+    )
+}
+
+fn gen_ffi_docs(struct_name: &Ident, derive_method_name: &syn::Ident) -> String {
+    format!(
+        " FFI function equivalent of [`{}::{}`]",
```

Comment:
```suggestion
        "FFI function equivalent of [`{}::{}`]",
```
Or I may be wrong, but I don't think that the single space is intentional.

Reply:
it's intentional because this without the leading space, the `#doc` macro is expanded into:
```
///FFI function equivalent of [`{}::{}`]"
```

Reply:
Good to know. I'd add a comment about it.